utils/mgmt_system/kubernetes.py

```diff
@@ -0,0 +1,167 @@
+from utils.rest_client import ContainerClient
+from collections import namedtuple
+
+"""
+Related yaml structures:
+
+[cfme_data]
+management_systems:
+    kubernetes:
+        name: My kubernetes
+        type: kubernetes
+        hostname: 10.12.13.14
+        port: 6443
+        credentials: kubernetes
+        authenticate: true
+        rest_protocol: https
+
+[credentials]
+kubernetes:
+    username: admin
+    password: secret
+    token: mytoken
+"""
+
+# Below are simplified representations of CFME k8s objects that should cover most of our needs
+Container = namedtuple('Container', ['name', 'cg_name', 'image'])
+Image = namedtuple('Image', ['name', 'tag', 'id'])
+# Images are equal if their ids match
+Image.__eq__ = lambda self, other: self.id == other.id
+ImageRegistry = namedtuple('ImageRegistry', ['host', 'port'])
+Project = namedtuple('Project', ['name'])
+Node = namedtuple('Node', ['name', 'ready', 'cpu', 'memory'])
+ContainerGroup = namedtuple(
+    'ContainerGroup', ['name', 'project_name', 'restart_policy', 'dns_policy'])
+ReplicationController = namedtuple(
+    'ReplicationController', ['name', 'project_name', 'replicas', 'current_replicas'])
+Service = namedtuple('Service', ['name', 'project_name', 'portal_ip', 'session_affinity'])
+
+
+class KubernetesException(BaseException):
```

Comment:
Subclassing BaseException is a [bad idea](https://docs.python.org/2/library/exceptions.html#exceptions.BaseException), and this exception should be defined in `utils.mgmt_system.exceptions`.